src/simulation/Simulation.h

```diff
@@ -47,9 +47,11 @@ class Simulation
     void setCurrentVirtualTime(VirtualClock::time_point t);
     void setCurrentVirtualTime(VirtualClock::system_time_point t);
 
-    // Add new node to the simulation. This function does not start the node.
-    // Callers are expected to call `start` or `startAllNodes` manually.
-    Application::pointer addNode(SecretKey nodeKey, SCPQuorumSet qSet,
+    // Add new node to the simulation. `cfg.QUORUM_SET` is not modified if
+    // `qSet` is `nullopt`.  This function does not start the node.  Callers are
+    // expected to call `start` or `startAllNodes` manually.
+    Application::pointer addNode(SecretKey nodeKey,
+                                 std::optional<SCPQuorumSet> qSet,
```

Comment:
> I think I'll end up reverting this because I'm going to try to rewrite the test to avoid using `Simulation`.

This is exactly what I wound up doing.

Reply:
Ah nice, glad we ended up not needing any changes in Simulation.